### DISCUSSION TITLE
npm包替换table模块

### DISCUSSION THREAD

**Discussion by zzxming** [opentiny/tiny-editor#70] [General]
现在的 table 模块基本的功能算是完成，但是还有对于一些情况还存在问题，我自己写了一个 [table module](https://github.com/zzxming/quill-table-up)，最近升级到 2.x 支持了，有完整 ts 且包含了单元测试，大致的 table 功能是差不多的，我想在这个项目中是不是能用 npm 包替换掉现在的 table 模块，也省下一些时间修复和重构。 @kagol 

如果是希望继续使用当前的 table 模块的话，我会把我发现的一些 table 的问题发一个 issue 作为待解决问题

**Comment by kagol** [comment]
@zzxming 可以的呀，替换之后我也多测试下，确保不影响原来的功能。

目前看到缺失的功能就是
1. 表格拖选单元格的模块(table-selection)支持Shift多选、左上角/右下角小圆点拖拽多选；
2. 选择的单元格支持通过右键菜单进行复制；
3. 行列外部的工具栏，可以实现单行、单列全选，类似excel（行高、列宽改变我看已经有了）

可以考虑迁移过去，不过不是核心功能，不紧急。

**Reply by zzxming** [reply]
左上角/右下角小圆点拖拽多选：这个功能其实就是点击单元格鼠标移动选择，功能上有点/重复了，确实需要这种选择方式吗

其他的我可以进行补充，之后有空会开一个pr进行功能替换

**Reply by kagol** [reply]
> 左上角/右下角小圆点拖拽多选：这个功能其实就是点击单元格鼠标移动选择，功能上有点/重复了，确实需要这种选择方式吗
> 
> 其他的我可以进行补充，之后有空会开一个pr进行功能替换

左上角/右下角小圆点拖拽多选：这个功能其实就是点击单元格鼠标移动选择，功能上有点/重复了，确实需要这种选择方式吗？确实已经有了拖选，没必要这个小圆点了

**Comment by kagol** [comment]
设置表格单元格背景色，还是希望能有一些常见的颜色，用户可以配置，然后可以增加一个色盘，用户可以自己在色盘选择任意颜色，并且可以输入颜色值。

类似以下这种，但是可以更加简化一些。

![image](https://github.com/user-attachments/assets/7573caa2-b526-46da-ab66-a8a6d185c132)

**Comment by kagol** [comment]
@zzxming 后续也可以考虑将 quill-table-up 项目捐赠到 OpenTiny 组织下，我们可以一起运营起来。

另外想了解下：[quill-table-up](https://github.com/zzxming/quill-table-up) 和 [quill-table](https://github.com/zzxming/quill-table) 这两个有什么区别呢？为什么要维护两个呢？

**Reply by zzxming** [reply]
quill-table-up 最开始只是实现 quill2.x 原生 table 的便捷使用，后续才改成 table 的加强模块。

quill-table 的 npm 包名是 quill1.3.7-table-module，名字不适合做版本升级，而且 quill2.x 的 api 修改有点影响功能，所以才另开的 quill-table-up 作为专门 2.x 版本的模块。
